src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -247,7 +248,7 @@ public SessionFactoryImpl(Configuration cfg, IMapping mapping, Settings settings
 					if (cache != null)
 					{
 						caches.Add(cacheRegion, cache);
-						allCacheRegions.Add(cache.RegionName, cache.Cache);
+						((IDictionary<string, ICache>)allCacheRegions).Add(cache.RegionName, cache.Cache);
```

Comment:
@stefan-steinegger I had that same concern too, but I looked at the [Microsoft Reference Source](http://referencesource.microsoft.com/#mscorlib/system/Collections/Concurrent/ConcurrentDictionary.cs,1218) and the `IDictonary` implementation is a very simple wrapper around the main `TryAdd` and it also throws if not added (`TryAdd` returns `false`).  This throw code could easily be moved into the application, but since it is in the library, in the explicit `IDictionary` definitions, I left it there.